File: ios/ClearNews/ClearNews/ClearNews/Models/Article.swift
import Foundation

struct Article: Codable, Identifiable, Hashable {
    let title: String
    let summary: String?
    let url: String
    let imageUrl: String?
    let sourceId: String
    let sourceName: String
    let sourceType: String
    let category: String
    let sentiment: Double?
    let publishedAt: Date?

    var id: String { url }
}

struct ArticleListResponse: Codable {
    let articles: [Article]
    let pagination: Pagination
}

struct Pagination: Codable {
    let page: Int
    let perPage: Int
    let total: Int
    let totalPages: Int
}

File: android/ClearNews/app/src/main/java/com/nikhilsi/clearnews/model/Article.kt
package com.nikhilsi.clearnews.model

import kotlinx.serialization.SerialName
import kotlinx.serialization.Serializable

@Serializable
data class Article(
    val title: String,
    val summary: String? = null,
    val url: String,
    @SerialName("image_url") val imageUrl: String? = null,
    @SerialName("source_id") val sourceId: String,
    @SerialName("source_name") val sourceName: String,
    @SerialName("source_type") val sourceType: String,
    val category: String,
    val sentiment: Double? = null,
    @SerialName("published_at") val publishedAt: String? = null,
)

@Serializable
data class Pagination(
    val page: Int,
    @SerialName("per_page") val perPage: Int,
    val total: Int,
    @SerialName("total_pages") val totalPages: Int,
)

@Serializable
data class ArticleListResponse(
    val articles: List<Article>,
    val pagination: Pagination,
    val complete: Boolean? = null,
)
